edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/handlers/TelemetryHandler.cs

```diff
@@ -59,16 +55,16 @@ async Task<bool> HandleTelemetry(Match match, MqttPublishInfo publishInfo)
                                 ? this.identityProvider.Create(id1.Value, id2.Value)
                                 : this.identityProvider.Create(id1.Value);
 
-            var maybeProxy = await this.connectionRegistry.GetDeviceListenerAsync(identity);
-            var proxy = default(IDeviceListener);
+            var maybeListener = await this.connectionRegistry.GetDeviceListenerAsync(identity);
+            var listener = default(IDeviceListener);
 
             try
             {
-                proxy = maybeProxy.Expect(() => new Exception($"No device listener found for {identity.Id}"));
+                listener = maybeListener.Expect(() => new Exception($"No device listener found for {identity.Id}"));
```

Comment:
It’s the exception from expect caught right away by the try catch? It seems that you can just log and return false directly

Reply:
I need to extract the value from an Option, and because we don't have unwrap(), I need to call Expect(). The other way would be use something like GetOrDefault() and then check for null